lightyear/src/transport/webtransport/client_wasm.rs

```diff
@@ -0,0 +1,176 @@
+#![cfg(target_family = "wasm")]
+//! WebTransport client implementation.
+use super::MTU;
+use crate::transport::{PacketReceiver, PacketSender, Transport};
+use anyhow::Context;
+use bevy::tasks::{IoTaskPool, TaskPool};
+use std::net::SocketAddr;
+use tokio::select;
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::error::TryRecvError;
+use tracing::{debug, error, info, trace};
+use web_sys::js_sys::{Array, Uint8Array};
+use web_sys::wasm_bindgen::JsValue;
+use web_sys::WebTransportHash;
+
+use base64::prelude::{Engine as _, BASE64_STANDARD};
+use xwt_core::prelude::*;
+use xwt_web_sys::{Connection, Endpoint};
+
+/// WebTransport client socket
+pub struct WebTransportClientSocket {
+    client_addr: SocketAddr,
+    server_addr: SocketAddr,
+    certificate_digest: String,
+}
+
+impl WebTransportClientSocket {
+    pub fn new(
+        client_addr: SocketAddr,
+        server_addr: SocketAddr,
+        certificate_digest: String,
+    ) -> Self {
+        Self {
+            client_addr,
+            server_addr,
+            certificate_digest,
+        }
+    }
+}
+
+fn js_array(values: &[&str]) -> JsValue {
+    return JsValue::from(
+        values
+            .into_iter()
+            .map(|x| JsValue::from_str(x))
+            .collect::<Array>(),
+    );
+}
+
+impl Transport for WebTransportClientSocket {
+    fn local_addr(&self) -> SocketAddr {
+        self.client_addr
+    }
+
+    fn listen(self) -> (Box<dyn PacketSender>, Box<dyn PacketReceiver>) {
+        let client_addr = self.client_addr;
+        let server_addr = self.server_addr;
+        let (to_server_sender, mut to_server_receiver) = mpsc::unbounded_channel();
+        let (from_server_sender, from_server_receiver) = mpsc::unbounded_channel();
```

Comment:
This can exhaust all available memory unless there is some other way to limit the amount of in-flight data in place

Reply:
In fact, a channel with no buffer might be useful here; `tokio` doesn't have these at all, but `flume` with size `0` is essentially that.

Reply:
I am adding a bandwidth cap in another PR, but I think that's a good callout; i'll add a TODO